cmake/version.h.in

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+// clang-format off
+#define CELERITY_VERSION_MAJOR @CELERITY_VERSION_MAJOR@
```

Comment:
Even though all of the defines in this file are version-related, we're mixing two different things here:

- Constants that are generated pre-build (Git revision) and have to go through the custom build step. These probably have to be in a header.
- Constants that are generated during CMake configuration (Version triple). These behave like `CELERITY_SYCL_IMPL`, `CELERITY_FEATURE_*` etc., which we currently pass as compiler flags instead.

Any thoughts on unifying the latter either in a generic `config.h` header or by passing the version triple as compiler flags?

Reply:
Hmm, I think there is value in having a header that contains this information in a Celerity installation (i.e. is greppable) as opposed to having it passed as a flag, as well as having it in a file called `version.h` since that seems to be a pretty common convention at this point. That being said, I like the idea of a separate `config.h` for everything else!

Reply:
Right, that's an important point.

Reply:
This is now properly separated into `version.h` and `version.cc`.